rfcs/065-library-data-link-explorer/README.md

```diff
@@ -0,0 +1,129 @@
+
+This RFC outlines the plan for the Library Data Link Explorer web application. This tool will enable Collections Information colleagues the ability to view and debug work relationships independently, potentially replacing the workflow of requesting a developer-run script to produce a matcher graph .dot file. 
+
+![Example DOT Notation Graph](https://github.com/user-attachments/assets/c3cdc4f2-6c68-4b82-90c0-dea8da8cc0e5)
+
+*Example .dot graph returned by Matcher logic.*
+
+
+**The application will consist of:**
+
+- An API providing an endpoint that utilises existing matcher logic which returns a JSON graph structure for a given work ID.
+- A frontend interface that allows users to input a work ID and displays either a graph of connected works or an error message if no relations exist.
+
+### Background Context
+
+The catalogue data on [wellcomecollection.org](https://wellcomecollection.org/search) is created from multiple source systems. Different records in various source systems may refer to a single cultural artefact (referred to as a "work"), and one of the functions of the [catalogue pipeline](https://github.com/wellcomecollection/catalogue-pipeline) is to identify links between these source records and to merge them into single works.
+
+Records may be linked implicitly by sharing common identifiers, but more often the linking data is manually added and managed by colleagues in collections information (CI). This usually takes the form of an identifier for one system being stored in a particular field in another. This can lead to rich and complex tangles of records where the merging logic sometimes leads to results that are unexpected or even non-deterministic.
+
+However, CI colleagues currently have no visibility over how records are linked other than by manually inspecting individual records. Since 2021, our team has maintained a script for visualising the way records are linked (we call these "matcher graphs" after the name of the pipeline service which identifies the links), but this script can only be run by software engineers due to the AWS permissions it requires. Since the source information belongs to Collections Information, we would like them to be able to view these graphs (nodes and edges) themselves.
+
+### How will it work?
+
+
+1. **User inputs a work ID**: A Collections Information staff member enters an work ID into the application.
+2. **API fetches the data**: The API queries Elasticsearch for records linked to the work ID.
+3. **Frontend displays the graph**: The frontend renders a visual graph of nodes (works) and edges (relationships), allowing users to explore the connections using a graph library.
+4. **Enhanced Collections Information workflow**: This allows Collections Information colleagues to independently view and explore relationships between records without requiring software engineer assistance, boosting productivity and collaboration.
+
+---
+
+## Backend
+
+- An API endpoint, powered by AWS Lambda to:
```

Comment:
Yes - I'll use a FURL to expose the API endpoint.

I'll be sure to sanitise inputs to prevent injection attacks, throttle requests using Lambda concurrency limits. I'd also like to look at auth is there is time.